src/app/pages/client/explore/Explore.tsx

```diff
@@ -93,79 +109,181 @@ export function AddServer() {
                 <Text priority="400">Add server name to explore public communities.</Text>
                 <Box direction="Column" gap="100">
                   <Text size="L400">Server Name</Text>
                   <Input ref={serverInputRef} name="serverInput" variant="Background" required />
-                  {exploreState.status === AsyncStatus.Error && (
+                  {viewState.status === AsyncStatus.Error && (
                     <Text style={{ color: color.Critical.Main }} size="T300">
                       Failed to load public rooms. Please try again.
                     </Text>
                   )}
                 </Box>
                 <Box direction="Column" gap="200">
-                  {/* <Button
+                  <Button
                     type="submit"
+                    onClick={handleView}
                     variant="Secondary"
+                    fill="Soft"
                     before={
-                      exploreState.status === AsyncStatus.Loading ? (
+                      viewState.status === AsyncStatus.Loading && (
                         <Spinner fill="Solid" variant="Secondary" size="200" />
-                      ) : undefined
+                      )
                     }
-                    aria-disabled={exploreState.status === AsyncStatus.Loading}
+                    disabled={busy}
                   >
-                    <Text size="B400">Save</Text>
-                  </Button> */}
-
-                  <Button type="submit" onClick={handleView} variant="Secondary" fill="Soft">
                     <Text size="B400">View</Text>
                   </Button>
+                  <Button
+                    type="submit"
+                    onClick={handleSaveView}
+                    variant="Primary"
+                    fill="Soft"
+                    before={
+                      saveViewState.status === AsyncStatus.Loading && (
+                        <Spinner fill="Solid" variant="Secondary" size="200" />
+                      )
+                    }
+                    disabled={busy}
+                  >
+                    <Text size="B400">Save & View</Text>
+                  </Button>
                 </Box>
               </Box>
             </Dialog>
           </FocusTrap>
         </OverlayCenter>
       </Overlay>
-      <Button
-        variant="Secondary"
-        fill="Soft"
-        size="300"
-        before={<Icon size="100" src={Icons.Plus} />}
-        onClick={() => setDialog(true)}
-      >
-        <Text size="B300" truncate>
-          Add Server
-        </Text>
-      </Button>
+      <NavItem variant="Background" aria-selected={selected}>
+        <NavButton onClick={() => setDialog(true)}>
+          <NavItemContent>{children}</NavItemContent>
+        </NavButton>
+      </NavItem>
     </>
   );
 }
 
+type ExploreServerNavItemAction = {
+  onClick: () => Promise<void>;
+  icon: IconSrc;
+  alwaysVisible: boolean;
+};
```

Comment:
passing subcomponent props through parent component is wrong design. The entire component should have been passed.